python/cudf/cudf/tests/test_dataframe.py

```diff
@@ -7143,14 +7144,24 @@ def test_cudf_arrow_array_error():
         sr.__arrow_array__()
 
 
-@pytest.mark.parametrize("n", [0, 2, 5, 10, None])
-@pytest.mark.parametrize("frac", [0.1, 0.5, 1, 2, None])
+@pytest.mark.parametrize("n", [0, 2, 10, None])
+@pytest.mark.parametrize("frac", [0.3, 2, None])
 @pytest.mark.parametrize("replace", [True, False])
 @pytest.mark.parametrize("axis", [0, 1])
-def test_dataframe_sample_basic(n, frac, replace, axis):
-    # as we currently don't support column with same name
+def test_dataframe_sample_basic(
+    n, frac, replace, axis, random_state_tuple, make_weights
+):
+    # TODO: decouple valid parameter set from invalid parameter set and
+    # write two separate tests to reduce function complexity.
```

Comment:
Is this something you want to push to future work?

Reply:
These tests are very convoluted...

Reply:
Yes, and probably part of the reason why we have to create these more complicated fixtures to address for them. I can try to tackle this in this PR.

Reply:
If you feel like that's a blocker.

Reply:
In fact this might be a blocker because apparently some tests on `axis==1` was not properly executed because the weights to sample from doesn't match the width of the frame, and these tests somehow didn't run at all. I wouldn't be surprised if there are more tests skipped like this. I'm taking a second look at these tests.

Reply:
I separated `axis=0` and `axis=1` into two separates test. Since `axis=1` support fewer random states, it really doesn't make sense to share the same fixture pattern used from `axis=0`. I thus split the fixtures into 2 while sharing same underlying code path for both test cases.

I also separates the valid parameter combo from the invalid ones as much as possible. There's one case where unsupported arguments had to do with some combo `replace`, `random_state` and `weights` that I need to explicitly skip in test. But it's also covered in a separate test (for exception messages).

I'm a little tempting to remove the test case in series since it provides the same code coverage to the dataframe test in `axis=0` case. Thoughts?